test_scripts/Smoke/commonSmokeApi.lua

```diff
@@ -0,0 +1,245 @@
+  return config.deviceMAC
+end
+
+function commonSmokeApi.getPathToSDL()
+  return config.pathToSDL
+end
+
+function commonSmokeApi.getMobileAppId(pAppId)
+  if not pAppId then pAppId = 1 end
+  return config["application" .. pAppId].registerAppInterfaceParams.appID
+end
+
+function commonSmokeApi.getSelfAndParams(...)
+  local out = { }
+  local selfIdx = nil
+  for i,v in pairs({...}) do
+    if type(v) == "table" and v.isTest then
+      table.insert(out, v)
+      selfIdx = i
+      break
+    end
+  end
+  local idx = 2
+  for i = 1, table.maxn({...}) do
+    if i ~= selfIdx then
+      out[idx] = ({...})[i]
+      idx = idx + 1
+    end
+  end
+  return table.unpack(out, 1, table.maxn(out))
+end
+
+function commonSmokeApi.getMobileSession(pAppId, self)
+  self, pAppId = commonSmokeApi.getSelfAndParams(pAppId, self)
+  if not pAppId then pAppId = 1 end
+  return self["mobileSession" .. pAppId]
+end
+
+function commonSmokeApi.activateApp(pAppId, self)
+  self, pAppId = commonSmokeApi.getSelfAndParams(pAppId, self)
+  if not pAppId then pAppId = 1 end
+  local pHMIAppId = hmiAppIds[config["application" .. pAppId].registerAppInterfaceParams.appID]
+  local mobSession = commonSmokeApi.getMobileSession(pAppId, self)
+  local requestId = self.hmiConnection:SendRequest("SDL.ActivateApp", { appID = pHMIAppId })
+  EXPECT_HMIRESPONSE(requestId)
+  mobSession:ExpectNotification("OnHMIStatus",
+    {hmiLevel = "FULL", audioStreamingState = "AUDIBLE", systemContext = "MAIN"})
+  commonTestCases:DelayedExp(commonSmokeApi.minTimeout)
+end
+
+function commonSmokeApi.start(pHMIParams, self)
+  self, pHMIParams = commonSmokeApi.getSelfAndParams(pHMIParams, self)
+  self:runSDL()
+  commonFunctions:waitForSDLStart(self)
+  :Do(function()
+      self:initHMI(self)
+      :Do(function()
+          commonFunctions:userPrint(35, "HMI initialized")
+          self:initHMI_onReady(pHMIParams)
+          :Do(function()
+              commonFunctions:userPrint(35, "HMI is ready")
+              self:connectMobile()
+              :Do(function()
+                  commonFunctions:userPrint(35, "Mobile connected")
+                  allowSDL(self)
+                end)
+            end)
+        end)
+    end)
+end
+
+function commonSmokeApi.registerApplicationWithPTU(pAppId, pUpdateFunction, self)
+  self, pAppId, pUpdateFunction = commonSmokeApi.getSelfAndParams(pAppId, pUpdateFunction, self)
+  if not pAppId then pAppId = 1 end
+  self["mobileSession" .. pAppId] = mobile_session.MobileSession(self, self.mobileConnection)
+  self["mobileSession" .. pAppId]:StartService(7)
+  :Do(function()
+      local corId = self["mobileSession" .. pAppId]:SendRPC("RegisterAppInterface",
+        config["application" .. pAppId].registerAppInterfaceParams)
+      EXPECT_HMINOTIFICATION("BasicCommunication.OnAppRegistered",
+        { application = { appName = config["application" .. pAppId].registerAppInterfaceParams.appName } })
+      :Do(function(_, d1)
```

Comment:
@AKalinich-Luxoft Bad names `d1`, `d2`, etc

Reply:
@AByzhynar renamed in https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/1697/commits/1f9a96a07f40b1051bd1cd0e5a3b77a979c40def